modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/MoveKBObjectMVCActionCommand.java

```diff
@@ -156,12 +167,72 @@ private void _errorMessage(
 			String message)
 		throws IOException {
 
+		hideDefaultErrorMessage(actionRequest);
+
 		JSONObject jsonObject = JSONUtil.put("errorMessage", message);
 
 		JSONPortletResponseUtil.writeJSON(
 			actionRequest, actionResponse, jsonObject);
 	}
 
+	private double _getNearestPriority(
+		double nextKBArticlePriority, double previousKBArticlePriority) {
+
+		if ((nextKBArticlePriority % 1) != 0) {
+			int roundedPriority = (int)Math.floor(nextKBArticlePriority);
+
+			if (roundedPriority > previousKBArticlePriority) {
+				return roundedPriority;
+			}
+		}
+		else if ((nextKBArticlePriority - 1) > previousKBArticlePriority) {
+			return nextKBArticlePriority - 1;
+		}
+
+		return (previousKBArticlePriority + nextKBArticlePriority) / 2;
+	}
```

Comment:
Can you rename the method to "_getPriorityMidpoint" or something like that? You're trying to find the midpoint closest to an integer if it exist (correct me if I'm wrong); The current name doen't really say anything about that.

Reply:
Why not compute the average first and then do all the checks? I think it looks way simpler:
```
double average = (prev + next) / 2;

if ((average % 1) == 0) {
    return average;
}

int floor = Math.floor(average);

if (floor > prev) {
    return floor;
}

int ceiling = Math.ceil(average);

if (ceiling < next) {
    return ceiling;
}

return average;
```

Reply:
Also, consider using `floor == ceiling` instead of `average % 1`; the behavior of `%` with floating point numbers is a bit obscure.